deploy/jbossas/modeshape-jbossas-subsystem/src/main/java/org/modeshape/jboss/subsystem/AddS3BinaryStorage.java

```diff
@@ -41,6 +41,8 @@ protected void writeBinaryStorageConfiguration( String repositoryName,
         binaries.setString(FieldName.USER_PASSWORD, password);
         String bucketName = ModelAttributes.S3_BUCKET_NAME.resolveModelAttribute(context, model).asString();
         binaries.setString(FieldName.BUCKET_NAME, bucketName);
+        String endPoint = ModelAttributes.S3_ENDPOINT_URL.resolveModelAttribute(context, model).asString();
```

Comment:
this won't work correctly if `ModelAttributes.S3_ENDPOINT_URL` is not defined (the `asString()` part). It will actually return `undefined` instead of `null`.

So the correct code should look like:
```
ModelNode node = resolveModelAttribute();
String endpoint = node.isDefined() ? node.asString() : null
```